### PR TITLE
Overall code cleanup

### PR DESCRIPTION
**Pull request by bsipocz**
This is to address #623.

### CODE REVIEW

File: astroquery/alma/core.py [hunk 1]
```diff
@@ -196,17 +196,6 @@ def stage_data(self, uids):
...
-        With log.set_level(10)
-        INFO: Staging files... [astroquery.alma.core]
-        DEBUG: First request URL: https://almascience.eso.org/rh/submission [astroquery.alma.core]
-        DEBUG: First request payload: {'dataset': [u'ALMA+uid___A002_X3b3400_X90f']} [astroquery.alma.core]
-        DEBUG: First response URL: https://almascience.eso.org/rh/checkAuthenticationStatus/3f98de33-197e-4692-9afa-496842032ea9/submission [astroquery.alma.core]
-        DEBUG: Request ID: 3f98de33-197e-4692-9afa-496842032ea9 [astroquery.alma.core]
-        DEBUG: Submission URL: https://almascience.eso.org/rh/submission/3f98de33-197e-4692-9afa-496842032ea9 [astroquery.alma.core]
-        .DEBUG: Data list URL: https://almascience.eso.org/rh/requests/anonymous/786823226 [astroquery.alma.core]
-        """
-
```
Comment (keflavich) [inline astroquery/alma/core.py]: Keep this around.  It is "debug" information, but may be necessary for future code maintenance.  The ALMA API frequently updates, and it may be important to have this record to understand future changes.  It could probably be put in a better place, but this really counts as "notes in the margin"...

File: astroquery/alma/core.py [hunk 2]
```diff
@@ -306,26 +295,6 @@ def stage_data(self, uids):
...
-        # self._staging_log['data_list_page'] = data_list_page
-        # data_list_page.raise_for_status()
-
-        # if 'Error' in data_list_page.text:
-        #     errormessage = staging_root.find(
-        #         'div', id='errorContent').string.strip()
-        #     raise RemoteServiceError(errormessage)
-
-        # tbl = self._parse_staging_request_page(data_list_page)
-
```
Comment (keflavich) [inline astroquery/alma/core.py]: This was an old implementation.  Could you remove this in a separate commit, stating "Remove commented out old ALMA staging implementation"?  Again, it may be needed in the future.

File: astroquery/alma/core.py [hunk 3]
```diff
@@ -961,50 +887,3 @@ def unique(seq):
 def filter_printable(s):
     """ extract printable characters from a string """
     return filter(lambda x: x in string.printable, s)
-
-
-def parse_frequency_support(frequency_support_str):
```
Comment (keflavich) [inline astroquery/alma/core.py]: Ah... deleted because it is duplicated in utils?

File: astroquery/ned/tests/data/query_references.xml [hunk 4]
```diff
@@ -14,7 +14,7 @@ under cooperative agreement AST0122449 with The Johns Hopkins University.
 </DESCRIPTION>
```
Comment (keflavich) [inline astroquery/ned/tests/data/query_references.xml]: test data should not be modified unless NED is really returning something different now.  Is it?  If not, best revert this change.  If so, maybe this belongs in a different PR?

File: astroquery/ned/tests/test_ned.py [hunk 5]
```diff
@@ -94,7 +94,6 @@ def test_get_references_async(patch_get):
     assert response['search_type'] == 'Reference'
 
 
-@pytest.mark.xfail(reason="astropy issue #1266")
```
Comment (keflavich) [inline astroquery/ned/tests/test_ned.py]: just to confirm, is astropy#1266 resolved?

**Comment by keflavich** [comment]
:+1: except where I have left comments

File: astroquery/ned/tests/data/query_references.xml (hunk 4 above)
Comment (bsipocz) [inline astroquery/ned/tests/data/query_references.xml]: Yes, I've double checked it and it returns int numbers without the dots.

File: astroquery/ned/tests/test_ned.py (hunk 5 above)
Comment (bsipocz) [inline astroquery/ned/tests/test_ned.py]: Yes, it seems so. Also all tests (remotes, too) pass for this.

File: astroquery/alma/core.py (hunk 1 above)
Comment (bsipocz) [inline astroquery/alma/core.py]: fair enough. I was a bit hesitant to do this in the alma module, as it's one of the best maintained one :)

File: astroquery/alma/core.py (hunk 2 above)
Comment (bsipocz) [inline astroquery/alma/core.py]: OK.

File: astroquery/alma/core.py (hunk 3 above)
Comment (bsipocz) [inline astroquery/alma/core.py]: Yes. Also updated ``approximate_primary_beam_sizes`` in utils, to have all the same parameters as the one deleted from core.

**Comment by bsipocz** [comment]
Thanks @keflavich for the review, I'll update the commits according to the comments. Also this is a WIP PR, so more changes will come, just wanted to see whether the tests pass before doing any more work on it.

File: astroquery/ned/tests/data/query_references.xml (hunk 4 above)
Comment (keflavich) [inline astroquery/ned/tests/data/query_references.xml]: OK, great, thanks.  Maybe amend the commit message to include "NED now returns valid XML" or something along those lines?  Astroquery has actually lead to a few upstream fixes of this sort in the past, in Vizier at least, not sure about NED...

**Comment by keflavich** [comment]
Alright, sounds good.  Let me know when it's ready for a final review.

File: astroquery/ned/tests/data/query_references.xml (hunk 4 above)
Comment (bsipocz) [inline astroquery/ned/tests/data/query_references.xml]: Done.

**Comment by keflavich** [comment]
@bsipocz ready to merge?

**Comment by bsipocz** [comment]
Almost :)

**Comment by bsipocz** [comment]
@keflavich - is this performance info still needed in the code? 
https://github.com/astropy/astroquery/blob/master/astroquery/vizier/core.py#L129 

Is this included, as they are supposed to be improved? If yes, wouldn't it be better to move it into an issue? Or maybe #525 is already related to this?

**Comment by keflavich** [comment]
@bsipocz yeah, I agree this is better suited to an issue.  Feel free to factor out the private timing info & make an issue of it.

**Comment by bsipocz** [comment]
@keflavich - This is ready to be reviewed and merged. I'm sure there are still a couple of places where similar changes could be made, but we can save them for the future ;)

File: astroquery/conftest.py [hunk 6]
```diff
@@ -20,25 +19,6 @@
...
 
-
-# pytest magic:
-# http://pytest.org/latest/plugins.html#_pytest.hookspec.pytest_configure
-# use pytest.set_trace() to interactively inspect config's features
-def pytest_configure(config):
-    if config.getoption('remote_data'):
-        pass
-    else:
-        turn_off_internet(verbose=config.option.verbose)
```
Comment (keflavich) [inline astroquery/conftest.py]: Why is this removed?   Is it handled elsewhere now?

File: astroquery/cosmosim/core.py [hunk 7]
```diff
@@ -641,7 +630,8 @@ def _starttime_dict(self):
                  for i in range(len(response_list))]
         self.starttime_dict = {}
         for i in range(len(soups)):
-            self.starttime_dict['{}'.format(completed_ids[i])] = '{}'.format(soups[i].find('uws:starttime').string)
+            self.starttime_dict['{}'.format(completed_ids[i])] = (
+                '{}'.format(soups[i].find('uws:starttime').string))
```
Comment (keflavich) [inline astroquery/cosmosim/core.py]: these could also be simplified to `str(completed_ids[i])` and `str(soups[i].find('uws:starttime').string)`, I suppose?  They're acceptable as is though.

**Comment by keflavich** [comment]
Looks great!  Thanks for cleaning out all the old astropy0.3 hacks.  Once you answer the question about `turn_off_internet`, go ahead and merge!

File: astroquery/conftest.py (hunk 6 above)
Comment (bsipocz) [inline astroquery/conftest.py]: As far as I understand this is taken care of in ``astropy.tests.pytest_plugins`` (implemented by you in https://github.com/astropy/astropy/pull/2044 :) ) . E.g. https://github.com/astropy/astropy/blob/master/astropy/tests/pytest_plugins.py#L140

```
    if not config.getoption('remote_data'):
        turn_off_internet(verbose=config.option.verbose)
```

Comment (bsipocz) [inline astroquery/conftest.py]: @keflavich - Having a second look, maybe we can also remove turn_off/on_internet from ``astroquery/utils/testing_tools.py`` as it also got merged upstream into ``astropy/tests/disable_internet.py`` and isn't used anywhere in the code anyway.

File: astroquery/cosmosim/core.py (hunk 7 above)
Comment (bsipocz) [inline astroquery/cosmosim/core.py]: Thanks for the heads up, there were quite many more of these. Doing str() is probably an overkill in many of the cases, but we need tests for the module before making more changes. I cannot make it work anyway, not even following the docs (that somehow fools the doctesting).

**Comment by bsipocz** [comment]
Tests pass so merging. The one remaining question is whether to remove the ``turn_off/on_internet`` functions, but that can be done in a future PR.
